File: 0190-Reverse Bits/0190-Reverse Bits.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t result = 0;
        int i = 0;
        while (n)
        {
            ++i;
            result = (result << 1) + n % 2;
            n >>= 1;
        }
        return result << (32 - i);
    }
};

File: 0101-0200/0190-Reverse Bits/0190-Reverse Bits.py
class Solution:
    # @param n, an integer
    # @return an integer
    def reverseBits(self, n):
        i = result = 0
        while n:
            result = (result << 1) + (n&1)
            n >>= 1
            i += 1
        return result << (32 - i)
